src/high-level/include/iDynTree/KinDynComputations.h

```diff
@@ -371,13 +491,37 @@ class KinDynComputations {
      */
     iDynTree::Transform getWorldTransform(const iDynTree::FrameIndex frameIndex);
 
+    /**
+     * Return the transform where the frame is the frame
+     * specified by frameIndex, and the reference frame is the world one
+     * (world_H_frame).
+     * @param world_p_frame contains the position component of world_H_frame.
+     * @param world_R_frame contains the rotation component of world_H_frame.
+     * @warning the Span and the MatrixView objects should point an already existing memory. Memory allocation and resizing cannot be achieved with this kind of objects.
+     * @return true if all went well, false otherwise.
+     */
+    bool getWorldTransform(const iDynTree::FrameIndex frameIndex,
+                           const iDynTree::Span<double>& world_p_frame,
+                           const iDynTree::MatrixView<double>& world_R_frame);
+
     /**
      * Version of getWorldTransform where the frame is specified by name.
      *
      * @note For real time code use the method that takes an integer.
      *
      */
-    iDynTree::Transform getWorldTransform(std::string frameName);
+    iDynTree::Transform getWorldTransform(const std::string & frameName);
+
+    /**
+     * Version of getWorldTransform where the frame is specified by name
+     * (MatrixView and Span implementation).
+     *
+     * @note For real time code use the method that takes an integer.
+     *
+     */
+    bool getWorldTransform(const std::string & frameName,
```

Comment:
Maybe: do you need to access spans of `Transform` objects?
If you have an immediate need for this (meaning: you have some code that MUST use the span here), then, forget about this comment.
Otherwise, I suggest to drop the span version of methods accepting/returning Transform objects. 
These objects are not directly mapped to matrices, and the resulting API  is ugly and confusing

Reply:
The idea is to be able to use Eigen-matrix only (or any other matrix library) to interface with KinDynComputations. In that case, you need in any case to transform for your specific representation of Transform (in our codebases almost always 4x4 homegenous matrix) having a span variant of the Transform methods permits to reduce to the minimum the boilerplate necessary to use iDynTree classes. If you think about it, it is a bit similar to what we do in WB-Toolbox, where when we need to have a Simulink signal that represent an iDynTree::Transform, we use a 4x4 matrix.

Reply:
Fine. 
But as you probably know (as you wrote the `Transform` class), the 4x4 is just one of the possible representations. And the class is kind of clear on this. 
Adding a span version adds confusion IMHO. 
If this is needed I suggest to change the name of the method at least, to make this explicit. 
For example `getWorldTransformComponents`

Reply:
Ahh, I see the main point here. I think on the other hand that having exactly the same method names is useful for migration, but indeed having the same name but a different list of arguments is indeed confusing. Could it make sense to have for methods with the same names the same list of arguments, so having in place of a `iDynTree::Transform`  a 4x4 MatrixView that represent the corresponding homogenous transform? Probably this applies to other parts of the API where we have a `iDynTree::Transform`, for example `setRobotState` .

Reply:
This is not a trivial decision.
What I mean is:
- I consider it a safe change (modulo your comments on my other comment) to change a parameter from a vector/array to a span-like object. Conversion should be either implicit or simple by helper functions. In this case I do not see a reason to change name (nor to keep the old functions around).
- This is a different case. You can have a version taking a 4x4 view, maybe extend `Transform` with a constructor taking a 4x4 matrix, but for users who wants to use the explicit `Transform` (and I see the value in it) it will be a 2-steps procedure, plus waisting an intermediary object. This is why I was suggesting making it clear that they are two different methods

Reply:
>  You can have a version taking a 4x4 view, maybe extend Transform with a constructor taking a 4x4 matrix, but for users who wants to use the explicit Transform (and I see the value in it) it will be a 2-steps procedure, plus waisting an intermediary object. This is why I was suggesting making it clear that they are two different methods

At the moment, people will continue to use the Transform variant as usual without any kind of 2-steps procedure, as we are going to leave both variant around. If we deprecate/remove the Transform version and just leave the span version (that now to be honest seems unlikely to me) we will address about this aspects. 

In a nutshell, I think your comment would make sense if we were going to remove the non-span/view versions soon, but as mention in the other thread for now we will keep both versions to see how hergonomic the use of the span/view version is.

Reply:
No I am not suggesting to remove the `Transform` version (while I was suggesting to remove the `Vector` version to the other methods).
If you change the span version to get a span to a 4x4 matrix, (and document that that is a homogenous transformation matrix) then it looks good to me :)